### ISSUE TITLE
Initialization error on resume from sleep

### ISSUE THREAD

**Issue by titan253** [medevil84/FanControl.AquacomputerDevices#3]
Hi there, the plug in operates without issues on a fresh startup but when resuming from sleep I always get an Initialize failed error. I run a refresh sensor detection and that usually will fix it. I've attached the error log for the most recent wake from sleep, do I need to set anything in Windows to avoid this? I do have FanControl set on a 15 second delay but I don't know if that apples to a resume from sleep.

Thanks!

```
5/19/2023 11:23:34 AM: System.ApplicationException: This operation returned because the timeout period expired. (Exception from HRESULT: 0x800705B4)
   at System.Threading.ReaderWriterLock.AcquireWriterLockInternal(Int32 millisecondsTimeout)
   at FanControl.AquacomputerDevices.Devices.QuadroDevice.Settings_ResetControllerPower(Int32 index)
   at FanControl.ViewModels.MainWindowViewModel.Clear()
   at System.Windows.Threading.DispatcherOperation.InvokeDelegateCore()
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Windows.Threading.DispatcherOperation.Wait(TimeSpan timeout)
   at System.Windows.Threading.Dispatcher.InvokeImpl(DispatcherOperation operation, CancellationToken cancellationToken, TimeSpan timeout)
   at System.Windows.Threading.Dispatcher.Invoke(Action callback, DispatcherPriority priority, CancellationToken cancellationToken, TimeSpan timeout)
   at System.Windows.Threading.Dispatcher.Invoke(Action callback)
   at FanControl.Domain.ComputerAccessLayer.Close()
   at FanControl.Domain.ComputerAccessLayer.<>c__DisplayClass11_0.<<TryRefresh>b__0>d.MoveNext()
5/19/2023 11:23:34 AM: Initialization failed.
5/19/2023 11:30:30 AM: AquacomputerPlugin: Closing.
5/19/2023 11:30:30 AM: QuadroDevice.Unload()
5/19/2023 11:30:30 AM: AquacomputerPlugin: Initializing.
5/19/2023 11:30:30 AM: QuadroDevice.AssignDevice(device: VendorID=0x0C70, ProductID=0xF00D, Version=256, DevicePath=\\?\hid#vid_0c70&pid_f00d&mi_01#b&322cee34&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}, logger: FanControl.Domain.PluginLogger)
5/19/2023 11:30:30 AM: AquacomputerPlugin: Loading
5/19/2023 11:30:30 AM: AquacomputerPlugin: Loading device FanControl.AquacomputerDevices.Devices.QuadroDevice
5/19/2023 11:30:30 AM: QuadroDevice.Load(_container: FanControl.Domain.BackendProviders.Plugin.PluginSensorsContainer)
```

**Comment by medevil84** [comment]
Hello, can you try replacing the dlls with the one attached in the zip file and tell me if the issue is still present? Also, please let me know if you see any strange behaviour.
[Debug.zip](https://github.com/medevil84/FanControl.AquacomputerDevices/files/11525325/Debug.zip)

**Comment by titan253** [comment]
Thanks so much! After testing for a few days I haven't encountered any issues resuming from sleep and haven't seen anything pop up in the error logs. 

Everything seems to be operating as intended, thanks again!

**Comment by medevil84** [comment]
Fixed in 1.0.0.4
